consensus/state_processing/src/per_epoch_processing/capella/full_withdrawals.rs

```diff
@@ -1,10 +1,23 @@
+use crate::common::withdraw_balance;
 use crate::EpochProcessingError;
-use types::beacon_state::BeaconState;
-use types::eth_spec::EthSpec;
+use types::{beacon_state::BeaconState, eth_spec::EthSpec, ChainSpec};
 
 pub fn process_full_withdrawals<T: EthSpec>(
-    _state: &mut BeaconState<T>,
+    state: &mut BeaconState<T>,
+    spec: &ChainSpec,
 ) -> Result<(), EpochProcessingError> {
-    todo!("implement this");
+    let current_epoch = state.current_epoch();
+    // FIXME: is this the most efficient way to do this?
+    for validator_index in 0..state.validators().len() {
```

Comment:
This looks very similar to what you need here: https://github.com/sigp/lighthouse/blob/stable/beacon_node/http_api/src/lib.rs#L547

You don't have to deal with the optional from `get` cause you're just iterating over arrays